Sources/JSONUtilities/Dictionary+JSONKey.swift

```diff
@@ -318,15 +256,48 @@ extension Dictionary where Key: StringProtocol {
     return jsonArray
   }
 
-  // MARK: JSONArray decoding
+  // MARK: Dictionary decoding
 
-  fileprivate func decodableObjectsArray<ReturnType: JSONObjectConvertible>(_ jsonArray: JSONArray) -> [ReturnType] {
-    return jsonArray.flatMap {
-      guard let castedJsonObject = $0 as? JSONDictionary else {
-        return nil
+  fileprivate func decodeDictionary<T>(atKeyPath keyPath: Key, invalidItemBehaviour: InvalidItemBehaviour = .remove, decodeClosure: (JSONDictionary, String) throws -> T?) throws -> [String: T] {
```

Comment:
Why does `decodeDictionary` take a `(JSONDictionary, String)` for decoding but `decodeArray` only takes `(Any)`, what am I missing? Is the `String` a `key` or a `keyPath`?